tilequeue/utils.py

```diff
@@ -23,3 +26,51 @@ def grouper(iterable, n):
         if not chunk:
             return
         yield chunk
+
+def parse_log_file(log_file):
+    ip_pattern = '(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'
+    # didn't match againts explicit date pattern, in case it changes
+    date_pattern = '\[([\d\w\s\/:]+)\]'
+    tile_id_pattern = '\/([\w]+)\/([\d]+)\/([\d]+)\/([\d]+)\.([\d\w]*)'
+
+    log_pattern = '%s - - %s "([\w]+) %s.*' % (ip_pattern, date_pattern, tile_id_pattern)
+
+    tile_log_records = []
+    for log_string in log_file:
+        match = re.search(log_pattern, log_string)
+        if match and len(match.groups()) == 8:
+            tile_log_records.append((match.group(1), 
+                                      datetime.strptime(match.group(2), '%d/%B/%Y %H:%M:%S'), 
+                                      coord_marshall_int(create_coord(match.group(6), match.group(7), match.group(5)))))
+
+    return tile_log_records
+
+def mimic_prune_tiles_of_interest_sql_structure(cursor):
+    cursor.execute('''CREATE TABLE IF NOT EXISTS tile_traffic_v4 (
+            id bigserial primary key,
+            date timestamp(6) not null,
+            z integer not null,
+            x integer not null,
+            y integer not null,
+            tilesize integer not null,
+            service varchar(32),
+            host inet not null
+        )''')
+
+def postgres_add_compat_date_utils(cursor):
```

Comment:
I didn't want to alter your query to redshift or have 2 distinct queries in prune_tiles_of_interest, since standard Postgres doesn't have dateadd() and getdate() I just added their implementation so there is no need for altering the query or differentiation between Postgres and redshift